markdown/developers/pipeline_structure.md

```diff
@@ -0,0 +1,74 @@
+# nf-core contribution guidelines
+
+Nf-core pipelines adhere to common design principles that have been crafted through discussions and extensive real-world experiences with pipelines over time. We understand, however, that not all about nf-core pipelines is obvious and can be confusing to people new to nf-core. To help you along getting started with nf-core, writing your pipeline or contribute otherwise, we have created this documentation. Hopefully you will find most things that might confuse you described here, and if not, be sure to bug the nf-core Slack channel!
```

Comment:
Minor minor super-minor things that I generally try to stick to when writing nf-core docs:

* Lower case `nf-core`, try to avoid starting a sentence with it because it looks weird
* I usually put each sentence on its own line in the md file. Doesn't affect rendering on the website and makes the diffs easier to read if people change things.

```suggestion
All nf-core pipelines adhere to common design principles that have been crafted through discussions and extensive real-world experiences with pipelines over time.
We understand, however, that not all about nf-core pipelines is obvious and can be confusing to people new to nf-core.
To help you along getting started with nf-core, writing your pipeline or contribute otherwise, we have created this documentation.
Hopefully you will find most things that might confuse you described here, and if not, be sure to bug the nf-core Slack channel!
```